src/lib/useKeyfileGenerator.ts

```diff
@@ -242,14 +253,56 @@ export const useSingleKeyfileGenerator = ({
       revision: point.keyRevisionNumber,
     });
 
-    if (Just.hasInstance(derivedSeed)) {
-      setSeeds([derivedSeed.value]);
+    if (Just.hasInstance(defaultDerivedSeed)) {
+      setDefaultSeeds([defaultDerivedSeed.value]);
+    }
+
+    /**
+     * Otherwise, we will need to try a different token derivation scheme.
+     *
+     * Due to a logical error with auth token derivation, keys set between May
+     * 2021 and March 2022 used `sha256` instead of the logically correct
+     * `keccak256`. To mitigate this for users who log back into Bridge to
+     * download their keyfile or copy their access code, we need to check for
+     * the existence by deriving with a fallback algorithm.
+     *
+     * See this issue for additional context:
+     * https://github.com/urbit/bridge/issues/549
+     */
+
+    if (NONCUSTODIAL_WALLETS.has(walletType)) {
+      console.log('cannot derive fallback token for non-custodial wallets');
+      return;
     }
-  }, [authMnemonic, authToken, point, seed, urbitWallet, wallet]);
+
+    const fallbackAuthToken = getSha256AuthToken({ wallet: wallet.value });
+    const fallbackDerivedSeed = deriveNetworkSeedFromAuthToken(
```

Comment:
Don't we want `attemptNetworkSeedDerivation` here too? We want this to fail if it doesn't match the on-chain keys.

Reply:
The on-chain check is happening in `useKeyfileGenerator#89`, which is downstream from this function. Will update the comments to reflect this.

Reply:
At this point in the flow, we still don't know which token could potentially be valid. The `keysMatchChain` downstream will perform the validation.